### DISCUSSION TITLE
Restart custom rsbuildServer when changed server side

### DISCUSSION THREAD

**Discussion by sowtame** [web-infra-dev/rsbuild#3566] [Q&A]
To begin with, thank you so much for the Multi-Environment Builds api!

But also it would be nice to have a native rsbuild api for cases, when the application can not only render some UI on the server, but may be have a small backend for frontend layer

Like in this example on every request load web bundle and render 
https://github.com/rspack-contrib/rspack-examples/blob/main/rsbuild/ssr-express/server.mjs#L27

and when you have more than one route for ssr, you need to create custom HMR handler. such as 
```
if (module.hot) {
    module.hot.accept(['./server'], async () => {
        try {
            await currentServer.close();
            await startServer();
        } catch (error) {
            console.log('Failed to update server. You probably need to restart application', error);
        }
    });
}
```
or something that https://github.com/sowtame/rsbuild-server-client-side/blob/main/server.mjs#L80

maybe some api when you can handle this logic by rsbuildServer api...

**Comment by 9aoy** [comment]
Thanks, we will investigate this feature.
